### PR TITLE
woodpecker-*: 0.15.11 -> 1.0.0

### LINKED ISSUE #246104: Update request: woodpecker 0.15.11 → 1.0.0
- Package name: `woodpecker-*`
- Latest released version: 0.15.8

- Current version on the unstable channel: 0.15.8
- Current version on the stable/release channel: 0.15.8
- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@ambroisie @techknowlogick

# Details

v1.0.0 [was released today](https://github.com/woodpecker-ci/woodpecker/releases/tag/v1.0.0), after a recent release candidate. There are a number of breaking changes, listed in the linked release notes. In particular, [the YAML config files have changed schema](https://github.com/woodpecker-ci/woodpecker/pull/1833) in a backward-incompatible way. However, this release does bring a number of welcome changes, e.g. [`when` evaluation filter expressions](https://github.com/woodpecker-ci/woodpecker/pull/1213), as well as a laundry list of bugfixes.

-----

Note for maintainers: Please tag this issue in your PR.

### PR DESCRIPTION
**Pull request by ambroisie**
## Description of changes

Closes #246104. Release note [here](https://github.com/woodpecker-ci/woodpecker/releases/tag/v1.0.0) and blog post [there](https://woodpecker-ci.org/blog/release-v1.0.0).

Since #231513 has not been closed, I've decided to bump `pnpm-lock-export` to a fork which has v6 support, in order to package the front-end.

This release includes a _lot_ of breaking changes.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [X] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ambroisie** [comment]
I have yet to test this on my infra, but uploading it now so that others can also try it out.

**Comment by tek** [comment]
I tried using the new `local` backend with this, and it complains that it can't find `plugin-git` in the `PATH`, even if it's in `systemPackages`.

**Comment by ambroisie** [comment]
@tek you need to add it to the service's path (it can access the system packages as a security measure).

See [my dotfiles](https://git.belanyi.fr/ambroisie/nix-config/src/branch/main/modules/services/woodpecker/agent-exec/default.nix) for an example.

File: pkgs/development/web/pnpm-lock-export/default.nix
Comment (ambroisie) [inline pkgs/development/web/pnpm-lock-export/default.nix:7]: Oops, this should be
```suggestion
    owner = "adamcstephens";
```

Comment (ambroisie) [reply pkgs/development/web/pnpm-lock-export/default.nix:7]: @adamcstephens will you try to open a PR for your patches?

Comment (adamcstephens) [reply pkgs/development/web/pnpm-lock-export/default.nix:7]: Done, though I'll note it's a different branch that's a bit cleaner for a PR: https://github.com/cvent/pnpm-lock-export/pull/9

**Comment by adamcstephens** [comment]
> I tried using the new local backend with this, and it complains that it can't find plugin-git in the PATH, even if it's in systemPackages.

Should we configure the module to add this to the service path by default?

**Comment by ambroisie** [comment]
> Should we configure the module to add this to the service path by default?

I think it's best not to bloat the closure of a service if it is not needed. As a middle-ground, I think we might be able to add a `path` option to the `services.woodpecker-agents.agents.<name>` attribute set (and make use of it in the example configuration).

**Comment by tek** [comment]
> @tek you need to add it to the service's path (it can't access the system packages as a security measure).
> 
> See [my dotfiles](https://git.belanyi.fr/ambroisie/nix-config/src/branch/main/modules/services/woodpecker/agent-exec/default.nix) for an example.

oh thanks, I examined the systemd attributes for quite a while and couldn't figure out what the right way to do this was!

**Review by adamcstephens — APPROVED** [review]
I've upgraded my server and agent to this PR, and verified the CLI.

I'll add myself as maintainer after merge unless you want to add me in this PR.

**Comment by ambroisie** [comment]
> I'll add myself as maintainer after merge unless you want to add me in this PR.

That's fine with me :-). I think you should be the author of the commit for adding yourself.

**Comment by pjjw** [comment]
hey thanks for doing this!

**Pull request merged**
